bee-common/bee-common/src/password.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides password utilities.
+
+use argon2::{self, Config};
+use rand::Rng;
+
+/// Generates a salt to be used for password hashing.
+pub fn generate_salt() -> [u8; 32] {
+    rand::thread_rng().gen::<[u8; 32]>()
+}
+
+/// Hashes a password together with a salt.
+pub fn password_hash(password: &[u8], salt: &[u8]) -> Result<Vec<u8>, String> {
```

Comment:
Why not return propagate the underlying `rust-argon2` error? By turning it into a string, you remove any real possibility of the caller deciding on a strategy to handle the error (or making the error message more useful for the end user). Also, it means unnecessary allocation.

Reply:
Yeah I asked myself about that but I somehow wanted to keep the possibility of switching the internal password lib without causing too much damage 🤔

Reply:
Well you could always just realias it locally